cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go

```diff
@@ -188,28 +191,51 @@ func resetConfigDir(configPathDir string, dirsToClean []string, isDryRun bool) {
 }
 
 // CleanDir removes everything in a directory, but not the directory itself
-func CleanDir(filePath string) error {
+func CleanDir(filePath string, dirsToBeRetain []string) error {
```

Comment:
i think we should keep CleanDir simple.
it is used for all directories during reset including /etc/kubernetes/pki, not only the kubelet dir.

the logic to check if a kubelet subdir is to be retained can be outside at the caller site of CleanDir.
we just need to stop calling CleanDir on the top level / whole kubelet dir, clean unrelated subdirs, but keep the mount dirs of failed unmounts.